clang/test/SemaSYCL/loop_fusion.cpp

```diff
@@ -0,0 +1,102 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -Wno-sycl-2017-compat -ast-dump -verify %s | FileCheck %s
```

Comment:
I think it's a bit cleaner to separate this into two test files: an ast dumping test in AST that contains no errors in the test cases and validates that the attributes are inherited as you'd expect, and a -verify test in SemaSYCL that validates the semantic uses of the attribute.